api/lock_api.go

```diff
@@ -0,0 +1,233 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// LockService is an API service which encapsulates the Git LFS Locking API.
+type LockService struct{}
+
+// Lock generates a *RequestSchema that is used to preform the "attempt lock"
+// API method.
+//
+// If a lock is already present, or if the server was unable to generate the
+// lock, the Err field of the LockResponse type will be populated with a more
+// detailed error describing the situation.
+//
+// If the caller does not have the minimum commit necessary to obtain the lock
+// on that file, then the CommitNeeded field will be populated in the
+// LockResponse, signaling that more commits are needed.
+//
+// In the successful case, a new Lock will be returned and granted to the
+// caller.
+func (s *LockService) Lock(req *LockRequest) (*RequestSchema, *LockResponse) {
+	var resp LockResponse
+
+	return &RequestSchema{
+		Method: http.MethodPost,
```

Comment:
I don't think `Method` should expect HTTP-specific values. The HTTP protocol implementation could just be an implementation detail even though I know that's the expected default implementation. It's just that dependency on HTTP-specific elements subtly scattered around is what made the `lfs` package more rigid than it needed to be and since you're refactoring...

How about making this a custom enum and just translating it to `http.MethodPost` in the http lifecycle?